src/dom/Value.ts

```diff
@@ -0,0 +1,126 @@
+import {Decimal, Timestamp, IonType} from "../Ion";
+import JSBI from "jsbi";
+import {Constructor} from "./DomValue";
+
+/**
+ * Provides a common interface to work with DOM elements of any Ion type.
+ *
+ * Implementors should return `null` for any method that cannot provide the requested value.
+ */
+export interface Value {
+    /**
+     * Returns the Ion type of this value.
+     */
+    getType(): IonType;
+
+    /**
+     * Returns an array containing any annotations associated with this value.
+     * If this value does not have any annotations, an empty array will be returned.
+     */
+    getAnnotations(): string[];
+
+    /**
+     * Returns true if this Value is a Null and false otherwise.
+     */
+    isNull(): boolean;
+
+    /**
+     * For the Boolean type, returns a JS boolean representation of the Value; otherwise
+     * throws an Error.
+     */
+    booleanValue(): boolean | null;
+
+    /**
+     * For the Integer, Float, and Decimal types, returns a JS number representation of the
+     * Value; otherwise throws an Error.
+     *
+     * In some cases, this method will perform conversions that inherently involve loss of
+     * precision. See bigIntValue() and decimalValue() for lossless alternatives.
+     */
+    numberValue(): number | null;
+
+    /**
+     * For the Integer type, returns a BigInt representation of the Value; otherwise throws
+     * an Error.
+     */
+    bigIntValue(): JSBI | null;
+
+    /**
+     * For the Decimal type, returns an ion.Decimal representation of the Value; otherwise
+     * throws an Error.
+     */
+    decimalValue(): Decimal | null;
+
+    /**
+     * For the String and Symbol types, returns a JS string representation of the Value's
+     * text; otherwise throws an Error.
+     */
+    stringValue(): string | null;
+
+    /**
+     * For the Timestamp type, returns a JS Date representation of the Value; otherwise throws
```

Comment:
Let's indicate that this may result in data loss

Reply:
Good call, added a note.